reducing_utils.go

```diff
@@ -9,10 +9,11 @@ Currently, only some functions are ported, the remaining will be ported soon.
 */
```

Comment:
we should rename this file.
in next PR.

Reply:
ok